src/main/scala/firrtl/transforms/ConstantPropagation.scala

```diff
@@ -160,7 +160,12 @@ class ConstantPropagation extends Transform with DependencyAPIMigration with Res
         case IntWidth(b) => b
       }
 
-      val v: Seq[Boolean] = s"%${w}s".format(a.value.toString(2)).map(_ == '1')
+      val maskedValue = if (a.value.signum < 0) {
+        a.value & ((BigInt(1) << w.toInt) - 1)
```

Comment:
Isn't it safe to eliminate the if/else and unconditionally perform this masking operation?